In [25]:
import warnings
warnings.simplefilter('ignore')

import pandas as pd

df = pd.read_csv('s3://ia-340-2025-fall-9874yay/Telemedicine_Use_in_the_Last_4_Weeks.csv')
df.head() 

Indicator              Group  \
0  Adults Who Had Appointment with Health Profess...  National Estimate   
1  Adults Who Had Appointment with Health Profess...             By Age   
2  Adults Who Had Appointment with Health Profess...             By Age   
3  Adults Who Had Appointment with Health Profess...             By Age   
4  Adults Who Had Appointment with Health Profess...             By Age   

           State       Subgroup  Phase  Time Period      Time Period Label  \
0  United States  United States    3.1           28  Apr 14 - Apr 26, 2021   
1  United States  18 - 29 years    3.1           28  Apr 14 - Apr 26, 2021   
2  United States  30 - 39 years    3.1           28  Apr 14 - Apr 26, 2021   
3  United States  40 - 49 years    3.1           28  Apr 14 - Apr 26, 2021   
4  United States  50 - 59 years    3.1           28  Apr 14 - Apr 26, 2021   

  Time Period Start Date Time Period End Date  Value  Low CI  High CI  \
0             04/14/2021           04/26/2021   25.7    25.0     26.4   
1             04/14/2021           04/26/2021   21.6    19.2     24.1   
2             04/14/2021           04/26/2021   23.1    21.7     24.5   
3             04/14/2021           04/26/2021   25.7    24.2     27.3   
4             04/14/2021           04/26/2021   26.3    24.6     28.1   

  Confidence Interval Quartile Range  Suppression Flag  
0         25.0 - 26.4            NaN               NaN  
1         19.2 - 24.1            NaN               NaN  
2         21.7 - 24.5            NaN               NaN  
3         24.2 - 27.3            NaN               NaN  
4         24.6 - 28.1            NaN               NaN

In [26]:
df.shape

(3344, 15)

In [29]:
df['Value']

0       25.7
1       21.6
2       23.1
3       25.7
4       26.3
        ... 
3339    18.8
3340    18.9
3341     NaN
3342    13.8
3343    10.7
Name: Value, Length: 3344, dtype: float64

In [31]:
df.sort_values(by='Time Period')

Indicator  \
1942  Households With Children Where Any Child Had A...   
1943  Households With Children Where Any Child Had A...   
1948  Households With Children Where Any Child Had A...   
1949  Households With Children Where Any Child Had A...   
1950  Households With Children Where Any Child Had A...   
...                                                 ...   
3321  Households With Children Where Any Child Had A...   
3322  Households With Children Where Any Child Had A...   
3323  Households With Children Where Any Child Had A...   
3276  Households With Children Where Any Child Had A...   
3324  Households With Children Where Any Child Had A...   

                                               Group          State  \
1942         By Health insurance of adult respondent  United States   
1943         By Health insurance of adult respondent  United States   
1948                       By Total household income  United States   
1949                       By Total household income  United States   
1950                       By Total household income  United States   
...                                              ...            ...   
3321                                        By State         Nevada   
3322                                        By State  New Hampshire   
3323                                        By State     New Jersey   
3276  By Race/Hispanic ethnicity of adult respondent  United States   
3324                                        By State     New Mexico   

                             Subgroup  Phase  Time Period  \
1942                     Private only   -1.0            1   
1943                       Any public   -1.0            1   
1948                $50,000 - $74,999   -1.0            1   
1949                $75,000 - $99,999   -1.0            1   
1950              $100,000 - $149,999   -1.0            1   
...                               ...    ...          ...   
3321                           Nevada    3.5           48   
3322                    New Hampshire    3.5           48   
3323                       New Jersey    3.5           48   
3276  Non-Hispanic Black, single race    3.5           48   
3324                       New Mexico    3.5           48   

          Time Period Label Time Period Start Date Time Period End Date  \
1942  Oct 12 - Nov 30, 2021             10/12/2021           11/30/2021   
1943  Oct 12 - Nov 30, 2021             10/12/2021           11/30/2021   
1948  Oct 12 - Nov 30, 2021             10/12/2021           11/30/2021   
1949  Oct 12 - Nov 30, 2021             10/12/2021           11/30/2021   
1950  Oct 12 - Nov 30, 2021             10/12/2021           11/30/2021   
...                     ...                    ...                  ...   
3321   Jul 27 - Aug 8, 2022             07/27/2022           08/08/2022   
3322   Jul 27 - Aug 8, 2022             07/27/2022           08/08/2022   
3323   Jul 27 - Aug 8, 2022             07/27/2022           08/08/2022   
3276   Jul 27 - Aug 8, 2022             07/27/2022           08/08/2022   
3324   Jul 27 - Aug 8, 2022             07/27/2022           08/08/2022   

      Value  Low CI  High CI Confidence Interval Quartile Range  \
1942    NaN     NaN      NaN                 NaN            NaN   
1943    NaN     NaN      NaN                 NaN            NaN   
1948    NaN     NaN      NaN                 NaN            NaN   
1949    NaN     NaN      NaN                 NaN            NaN   
1950    NaN     NaN      NaN                 NaN            NaN   
...     ...     ...      ...                 ...            ...   
3321   11.8     7.2     18.1          7.2 - 18.1       8.6-11.8   
3322   15.5     9.8     23.0          9.8 - 23.0      15.0-18.1   
3323   14.4     9.6     20.6          9.6 - 20.6      11.9-14.9   
3276   18.1    14.7     22.0         14.7 - 22.0            NaN   
3324   12.2     7.1     19.2          7.1 - 19.2      11.9-14.9   

      Suppression Flag  
1942               N

In [33]:
df[0:7]

Indicator              Group  \
0  Adults Who Had Appointment with Health Profess...  National Estimate   
1  Adults Who Had Appointment with Health Profess...             By Age   
2  Adults Who Had Appointment with Health Profess...             By Age   
3  Adults Who Had Appointment with Health Profess...             By Age   
4  Adults Who Had Appointment with Health Profess...             By Age   
5  Adults Who Had Appointment with Health Profess...             By Age   
6  Adults Who Had Appointment with Health Profess...             By Age   

           State       Subgroup  Phase  Time Period      Time Period Label  \
0  United States  United States    3.1           28  Apr 14 - Apr 26, 2021   
1  United States  18 - 29 years    3.1           28  Apr 14 - Apr 26, 2021   
2  United States  30 - 39 years    3.1           28  Apr 14 - Apr 26, 2021   
3  United States  40 - 49 years    3.1           28  Apr 14 - Apr 26, 2021   
4  United States  50 - 59 years    3.1           28  Apr 14 - Apr 26, 2021   
5  United States  60 - 69 years    3.1           28  Apr 14 - Apr 26, 2021   
6  United States  70 - 79 years    3.1           28  Apr 14 - Apr 26, 2021   

  Time Period Start Date Time Period End Date  Value  Low CI  High CI  \
0             04/14/2021           04/26/2021   25.7    25.0     26.4   
1             04/14/2021           04/26/2021   21.6    19.2     24.1   
2             04/14/2021           04/26/2021   23.1    21.7     24.5   
3             04/14/2021           04/26/2021   25.7    24.2     27.3   
4             04/14/2021           04/26/2021   26.3    24.6     28.1   
5             04/14/2021           04/26/2021   27.6    25.9     29.4   
6             04/14/2021           04/26/2021   28.7    26.9     30.5   

  Confidence Interval Quartile Range  Suppression Flag  
0         25.0 - 26.4            NaN               NaN  
1         19.2 - 24.1            NaN               NaN  
2         21.7 - 24.5            NaN               NaN  
3         24.2 - 27.3            NaN               NaN  
4         24.6 - 28.1            NaN               NaN  
5         25.9 - 29.4            NaN               NaN  
6         26.9 - 30.5            NaN               NaN

In [37]:
df.set_index('Time Period Start Date',inplace=True) 
df.head()

Indicator  \
Time Period Start Date                                                      
04/14/2021              Adults Who Had Appointment with Health Profess...   
04/14/2021              Adults Who Had Appointment with Health Profess...   
04/14/2021              Adults Who Had Appointment with Health Profess...   
04/14/2021              Adults Who Had Appointment with Health Profess...   
04/14/2021              Adults Who Had Appointment with Health Profess...   

                                    Group          State       Subgroup  \
Time Period Start Date                                                    
04/14/2021              National Estimate  United States  United States   
04/14/2021                         By Age  United States  18 - 29 years   
04/14/2021                         By Age  United States  30 - 39 years   
04/14/2021                         By Age  United States  40 - 49 years   
04/14/2021                         By Age  United States  50 - 59 years   

                        Phase  Time Period      Time Period Label  \
Time Period Start Date                                              
04/14/2021                3.1           28  Apr 14 - Apr 26, 2021   
04/14/2021                3.1           28  Apr 14 - Apr 26, 2021   
04/14/2021                3.1           28  Apr 14 - Apr 26, 2021   
04/14/2021                3.1           28  Apr 14 - Apr 26, 2021   
04/14/2021                3.1           28  Apr 14 - Apr 26, 2021   

                       Time Period End Date  Value  Low CI  High CI  \
Time Period Start Date                                                
04/14/2021                       04/26/2021   25.7    25.0     26.4   
04/14/2021                       04/26/2021   21.6    19.2     24.1   
04/14/2021                       04/26/2021   23.1    21.7     24.5   
04/14/2021                       04/26/2021   25.7    24.2     27.3   
04/14/2021                       04/26/2021   26.3    24.6     28.1   

                       Confidence Interval Quartile Range  Suppression Flag  
Time Period Start Date                                                       
04/14/2021                     25.0 - 26.4            NaN               NaN  
04/14/2021                     19.2 - 24.1            NaN               NaN  
04/14/2021                     21.7 - 24.5            NaN               NaN  
04/14/2021                     24.2 - 27.3            NaN               NaN  
04/14/2021                     24.6 - 28.1            NaN               NaN

In [45]:
df.loc['2021-12-05']

KeyError: '2021-12-05'

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3344 entries, 0 to 3343
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Indicator               3344 non-null   object 
 1   Group                   3344 non-null   object 
 2   State                   3344 non-null   object 
 3   Subgroup                3344 non-null   object 
 4   Phase                   3344 non-null   float64
 5   Time Period             3344 non-null   int64  
 6   Time Period Label       3344 non-null   object 
 7   Time Period Start Date  3344 non-null   object 
 8   Time Period End Date    3344 non-null   object 
 9   Value                   3104 non-null   float64
 10  Low CI                  3104 non-null   float64
 11  High CI                 3104 non-null   float64
 12  Confidence Interval     3104 non-null   object 
 13  Quartile Range          2142 non-null   object 
 14  Suppression Flag        52 non-null     

In [15]:
df.describe()

Phase  Time Period        Value       Low CI      High CI  \
count  3344.000000  3344.000000  3104.000000  3104.000000  3104.000000   
mean      3.019498    36.000598    19.605638    15.760374    24.118814   
std       0.990528    10.361604     5.368918     5.482188     5.889849   
min      -1.000000     1.000000     6.800000     3.800000    10.800000   
25%       3.100000    32.000000    15.800000    11.900000    19.900000   
50%       3.200000    37.500000    19.250000    15.400000    23.500000   
75%       3.400000    43.000000    23.000000    19.700000    27.500000   
max       3.500000    48.000000    44.900000    39.200000    57.400000   

       Suppression Flag  
count              52.0  
mean                1.0  
std                 0.0  
min                 1.0  
25%                 1.0  
50%                 1.0  
75%                 1.0  
max                 1.0

In [16]:
df = df.rename(columns={"Time Period Start Date": "Start Date", "Time Period End Date": "End Date"}) 
df.head()

Indicator              Group  \
0  Adults Who Had Appointment with Health Profess...  National Estimate   
1  Adults Who Had Appointment with Health Profess...             By Age   
2  Adults Who Had Appointment with Health Profess...             By Age   
3  Adults Who Had Appointment with Health Profess...             By Age   
4  Adults Who Had Appointment with Health Profess...             By Age   

           State       Subgroup  Phase  Time Period      Time Period Label  \
0  United States  United States    3.1           28  Apr 14 - Apr 26, 2021   
1  United States  18 - 29 years    3.1           28  Apr 14 - Apr 26, 2021   
2  United States  30 - 39 years    3.1           28  Apr 14 - Apr 26, 2021   
3  United States  40 - 49 years    3.1           28  Apr 14 - Apr 26, 2021   
4  United States  50 - 59 years    3.1           28  Apr 14 - Apr 26, 2021   

   Start Date    End Date  Value  Low CI  High CI Confidence Interval  \
0  04/14/2021  04/26/2021   25.7    25.0     26.4         25.0 - 26.4   
1  04/14/2021  04/26/2021   21.6    19.2     24.1         19.2 - 24.1   
2  04/14/2021  04/26/2021   23.1    21.7     24.5         21.7 - 24.5   
3  04/14/2021  04/26/2021   25.7    24.2     27.3         24.2 - 27.3   
4  04/14/2021  04/26/2021   26.3    24.6     28.1         24.6 - 28.1   

  Quartile Range  Suppression Flag  
0            NaN               NaN  
1            NaN               NaN  
2            NaN               NaN  
3            NaN               NaN  
4            NaN               NaN

In [17]:
df.isnull().sum()

Indicator                 0
Group                     0
State                     0
Subgroup                  0
Phase                     0
Time Period               0
Time Period Label         0
Start Date                0
End Date                  0
Value                   240
Low CI                  240
High CI                 240
Confidence Interval     240
Quartile Range         1202
Suppression Flag       3292
dtype: int64

In [18]:
df = df.dropna()
df.isnull().sum()

Indicator              0
Group                  0
State                  0
Subgroup               0
Phase                  0
Time Period            0
Time Period Label      0
Start Date             0
End Date               0
Value                  0
Low CI                 0
High CI                0
Confidence Interval    0
Quartile Range         0
Suppression Flag       0
dtype: int64

A) My dataset has 3344 rows, and 15 columns. There is a combination of numeric and text columns. My initial impression is the data is initially sorted by "Indicator", where they are either labled as "Adults Who Had Appointment with Health Professional Over Video or Phone, Last 4 Weeks" or "Households With Children Where Any Child Had Appointment with Health Professional Over Video or Phone, Last 4 Weeks".

B) I renamed the "Time Period Start Date" and "Time Period End Date" columns and shortend them to just "Start Date" and "End Date". There were 5454 missing values and I corrected this by using the drop.na() function. 

C) I selected the "Value" column and then sorted by "Time Period". The output indicates that the highest value was 25.7 and the lowest was 10.7. It also shows that the time period for the survey ranged from 1 to 48 days. 